libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/ranker/SimpleWeightedSumDataLayoutStrategyScorer.java

```diff
@@ -40,16 +40,11 @@ public List<ScoredDataLayoutStrategy> scoreDataLayoutStrategies(
 
     for (DataLayoutStrategy dataLayoutStrategy : dataLayoutStrategies) {
       double normalizedCost = minMaxNormalize(dataLayoutStrategy.getCost(), minCost, maxCost);
-      double normalizedGain = minMaxNormalize(dataLayoutStrategy.getGain(), minGain, maxGain);
+      double discountedGain = discountedGain(dataLayoutStrategy);
+      double normalizedGain = minMaxNormalize(discountedGain, minGain, maxGain);
       ScoredDataLayoutStrategy normalizedDataLayoutStrategy =
           ScoredDataLayoutStrategy.builder()
-              .dataLayoutStrategy(
-                  DataLayoutStrategy.builder()
```

Comment:
was this redundant or intentional, as we needed to create a new copy of the object?

Reply:
Incoming object doesn't have score computed, afaict. This meta object we want to compute the score. So now incoming object score will be set